eskip/eskip_test.go

```diff
@@ -7,297 +7,174 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
 )
 
-func checkItems(t *testing.T, message string, l, lenExpected int, checkItem func(int) bool) bool {
-	t.Helper()
-	if l != lenExpected {
-		t.Error(message, "length", l, lenExpected)
-		return false
-	}
-
-	for i := 0; i < l; i++ {
-		if !checkItem(i) {
-			t.Error(message, "item", i)
-			return false
-		}
-	}
-
-	return true
-}
-
-func checkFilters(t *testing.T, message string, fs, fsExp []*Filter) bool {
-	t.Helper()
-	return checkItems(t, "filters "+message,
-		len(fs),
-		len(fsExp),
-		func(i int) bool {
-			return fs[i].Name == fsExp[i].Name &&
-				checkItems(t, "filter args",
-					len(fs[i].Args),
-					len(fsExp[i].Args),
-					func(j int) bool {
-						return fs[i].Args[j] == fsExp[i].Args[j]
-					})
-		})
-}
-
-func TestParseRouteExpression(t *testing.T) {
+func TestParse(t *testing.T) {
 	for _, ti := range []struct {
 		msg        string
 		expression string
-		check      *Route
-		err        bool
+		check      []*Route
```

Comment:
minor: maybe expectedRoutes would be better naming?

Reply:
I did not rename struct fields to keep diff small. All three test use different naming